ui_testing/testcases/basic_tests/test05_testplans.py

```diff
@@ -51,8 +50,594 @@ def test001_test_plan_delete_testunit(self):
+        self.base_selenium.LOGGER.info('Asserting that the iterations of the testunit in the second testplan is the same as the updated iterations')
+        self.base_selenium.LOGGER.info("Asserting the iterations of the second testplan are: {}".format(new_iteration))
+        self.assertEqual(first_testunit_data_in_second_testplan['iterations'], int(new_iteration))
+
+    @skip('https://modeso.atlassian.net/browse/LIMS-6405')
+    def test016_delete_used_testplan(self):
+        '''
+        LIMS-3509
+        If a testplan is used, it can't be deleted
+        '''
+        test_plan_dict = self.get_active_article_with_tst_plan(test_plan_status='complete')
+        testplan_name = test_plan_dict['Test Plan Name']
+        testplan_article = test_plan_dict['Article Name']
+        testplan_materialtype = test_plan_dict['Material Type']
+    
+        # create a new order with this testplan
+        self.order_page.get_orders_page()
+        self.order_page.create_new_order(material_type=testplan_materialtype, article=testplan_article, test_plans=[testplan_name])
+
+        # delete testplan
+        self.test_plan.get_test_plans_page()
+        self.base_selenium.LOGGER.info('Testplan number: {} will be archived'.format(testplan_name))
+        testplan_deleted = self.test_plan.delete_selected_item_from_active_table_and_from_archived_table(item_name=testplan_name)
+
+        # check for the error popup that this testplan is used and can't be deleted
+        self.assertFalse(testplan_deleted)
+
+    def test017_archived_testplan_shouldnot_appear_in_order(self):
+        '''
+        LIMS-3708
+        In case a testplan is archived, it shouldn't appear when creating a new order
+        '''
+
+        # choose a random testplan
+        main_testplan_data = (self.test_plan.select_random_table_row(element='test_plans:test_plans_table'))
+
+        # get testplan data from an api call
+        testplan_data = (self.test_plan_api.get_testplan_with_filter(filter_option='number', filter_text=str(main_testplan_data['Test Plan No.'])))[0]
+
+        # get information, material type and article
+        testplan_name = testplan_data['testPlanName']
+        testplan_materialtype = testplan_data['materialType']
+        testplan_article = (testplan_data['article'])[0]
+
+        # archive this testplan
+        self.test_plan.archive_selected_items()
+
+        # go to order's section
+        self.order_page.get_orders_page()
+
+        # create a new order with material type and article same as the saved ones
+        order_data = self.order_page.create_new_order(material_type=testplan_materialtype, article=testplan_article, test_plans=[testplan_name])
+
+        # get the first suborder's testplan and make sure it's an empty string
+        suborder_first_testplan = (((order_data['suborders'])[0])['testplans'])[0]
+        self.assertEqual(len(suborder_first_testplan), 0)
```

Comment:
it got stuck in selecting the article, however that it passed
```
2020-01-01 22:35:17.094 | INFO     | ui_testing.testcases.base_test:setUp:31 - Test case : test017_archived_testplan_shouldnot_appear_in_order
2020-01-01 22:35:26.130 | INFO     | api_testing.apis.base_api:info:58 - Get authorized api session.
2020-01-01 22:35:26.819 | INFO     | api_testing.apis.base_api:info:58 - session ID : eyJhbGciOi .....
2020-01-01 22:35:26.821 | INFO     | ui_testing.pages.login_page:login:10 - Login admin : admin.
2020-01-01 22:35:34.836 | INFO     | ui_testing.pages.testplans_page:get_test_plans_page:11 -  + Get test plans page.
2020-01-01 22:35:37.372 | INFO     | ui_testing.pages.base_pages:sleep_small:40 -  Small sleep.
2020-01-01 22:35:42.378 | INFO     | ui_testing.pages.base_pages:info:234 - select random row
2020-01-01 22:35:43.392 | INFO     | api_testing.apis.base_api:info:58 - GET : https://automation.1lims.com/api/testPlans
2020-01-01 22:35:44.062 | INFO     | api_testing.apis.base_api:info:58 - Status code: 200
2020-01-01 22:35:44.390 | INFO     | ui_testing.pages.base_pages:confirm_popup:62 - Confirming the popup
2020-01-01 22:35:56.801 | INFO     | ui_testing.pages.base_pages:sleep_small:40 -  Small sleep.
2020-01-01 22:36:01.802 | INFO     | ui_testing.pages.order_page:create_new_order:97 -  Create new order.
2020-01-01 22:36:01.802 | INFO     | ui_testing.pages.orders_page:click_create_order_button:16 - Press create order button
2020-01-01 22:36:02.100 | INFO     | ui_testing.pages.base_pages:sleep_small:40 -  Small sleep.
2020-01-01 22:36:07.104 | INFO     | ui_testing.pages.order_page:set_new_order:13 - Set new order.
2020-01-01 22:36:11.756 | INFO     | ui_testing.pages.base_pages:sleep_small:40 -  Small sleep.
2020-01-01 22:36:18.925 | INFO     | ui_testing.pages.base_selenium:select_item_from_drop_down:376 -  There is no drop-down options
2020-01-01 22:36:21.331 | INFO     | ui_testing.pages.base_pages:sleep_small:40 -  Small sleep.
2020-01-01 22:36:30.703 | INFO     | ui_testing.pages.base_selenium:select_item_from_drop_down:391 -  There is no da0f4e8392 option in the drop-down
2020-01-01 22:36:30.703 | INFO     | ui_testing.pages.base_pages:sleep_small:40 -  Small sleep.
2020-01-01 22:36:35.779 | INFO     | ui_testing.pages.base_selenium:select_item_from_drop_down:356 -  Drop-down is disabled
2020-01-01 22:36:38.054 | INFO     | ui_testing.pages.base_pages:save:52 - save the changes
2020-01-01 22:36:48.167 | INFO     | ui_testing.pages.order_page:create_new_order:118 -  Order created with no : 1-20 
2020-01-01 22:36:48.291 | INFO     | ui_testing.pages.order_page:get_suborder_data:274 - getting main order data
2020-01-01 22:36:48.363 | INFO     | ui_testing.pages.order_page:get_suborder_data:281 - getting suborders data
2020-01-01 22:36:49.085 | INFO     | ui_testing.testcases.base_test:tearDown:52 - TearDown. 	
```

Reply:
@0xIslamTaha I don't get the issue, I tried it and it found an article and created the order successfully.

Reply:
@SalmaaRagab so when u ran it, it selected the article?

Reply:
@0xIslamTaha Yes.

Reply:
![image](https://user-images.githubusercontent.com/17981089/71687073-3aedcd80-2d9d-11ea-9eb2-94dbcc9790db.png)

![image](https://user-images.githubusercontent.com/17981089/71687079-4214db80-2d9d-11ea-831c-a9d543b215b7.png)

Reply:
@0xIslamTaha  ![image](https://user-images.githubusercontent.com/53828649/71723483-76d47180-2e35-11ea-8037-2e18f707f759.png)